pkg/controller/queuejob/queuejob_controller_ex.go

```diff
@@ -528,6 +558,21 @@ func (qjm *XController) GetQueueJobsEligibleForPreemption() []*arbv1.AppWrapper
 
 	if !qjm.isDispatcher { // Agent Mode
 		for _, value := range queueJobs {
+
+			if value.Status.State == arbv1.AppWrapperStateActive && value.Spec.SchedSpec.DispatchDuration.Limit > 0 {
+				awDispatchDurationLimit := value.Spec.SchedSpec.DispatchDuration.Limit
+				dispatchDuration := value.Status.ControllerFirstDispatchTimestamp.Add(time.Duration(awDispatchDurationLimit) * time.Second)
+				currentTime := time.Now()
+				hasDispatchTimeNotExceeded := currentTime.Before(dispatchDuration)
+
+				if !hasDispatchTimeNotExceeded {
+					klog.V(8).Infof("Appwrapper Dispatch limit exceeded, currentTime %v, dispatchTimeInSeconds %v", currentTime, dispatchDuration)
+					value.Spec.SchedSpec.DispatchDuration.Overrun = true
+					qjobs = append(qjobs, value)
+					//Got AW which exceeded dispatch runtime limit, move to next AW
+					continue
+				}
+			}
 			replicas := value.Spec.SchedSpec.MinAvailable
 
 			// Skip if AW Pending or just entering the system and does not have a state yet.
```

Comment:
I think line 579 should be at the top of the loop, right?

Reply:
Thanks, let me make that change